### DISCUSSION TITLE
Wait in Auto Materlization

### DISCUSSION THREAD

**Discussion by imhamadkhan** [dagster-io/dagster#18785] [Q&A]
Is there a way to skip or wait in auto materlization , if one of parent is in in progress while one is materlized. 
- i want to proceed materlization if anyone of parent is materlized , even though some parents are missing.
- But i want an assest to wait if some parent is in progress in auto materlization, to avoid redundant runs.

**Comment by OwenKephart** [comment]
Hi @imhamadkhan -- there are a number of changes currently in progress to help enable more flexibility here, so this answer may soon be out of date, but in the current world:

- Your first bullet is currently achievable as described here: https://docs.dagster.io/concepts/assets/asset-auto-execution#auto-materialize-even-if-some-parents-are-missing
- Your second bullet is not currently achievable, and the issue tracking this behavior is here: https://github.com/dagster-io/dagster/issues/16555
